tests/performance/reindexing/reindexing_and_feeding.rb

```diff
@@ -0,0 +1,255 @@
+
+# coding: utf-8
+# Copyright 2020 Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+require 'performance_test'
+require 'app_generator/search_app'
+
+class ReindexingAndFeeding < PerformanceTest
+
+  def initialize(*args)
+    super(*args)
+  end
+
+  def timeout_seconds
+    900
+  end
+
+  def setup
+	  super
+	  set_description("Measure throughput of reindexing, and its impact on external updates and puts")
+	  set_owner("jvenstad")
+	  @test_config = {
+		  {
+			  :legend => 'Reindexing',
+			  :fbench => { :clients => 4, :use_post => true },
+			  :data => "{ \"fields\": { \"text\": \"GNU's not UNIX\" } }"
+		  } => {
+			  'qps' => { :y_min =>  2650, :y_max =>  3250 },
+			  '95p' => { :y_min =>   1.4, :y_max =>   1.7 }
+		  },
+		  {
+			  :legend => 'GET small data 4 clients',
+			  :fbench => { :clients => 4, :use_post => false }
+		  } => {
+			  'qps' => { :y_min =>  5600, :y_max =>  6250 },
+			  '95p' => { :y_min =>   0.7, :y_max =>   0.9 }
+		  },
+		  {
+			  :legend => 'POST large data 128 clients',
+			  :fbench => { :clients => 128, :use_post => true},
+			  :data => "{ \"fields\": { \"text\": \"GNU#{"'s not UNIX" * (1 << 10) }\" } }"
+		  } => {
+			  'qps' => { :y_min =>  6300, :y_max =>  6700 },
+			  '95p' => { :y_min =>    24, :y_max =>    28 }
+		  },
+		  {
+			  :legend =>  'GET large data 128 clients',
+			  :fbench => { :clients => 128, :use_post => false }
+		  } => {
+			  'qps' => { :y_min => 29500, :y_max => 33000 },
+			  '95p' => { :y_min =>   4.9, :y_max =>   5.5 }
+		  }
+	  }
+	  @graphs = get_graphs
+	  @app = SearchApp.new.monitoring("vespa", 60).
+		  container(Container.new("combinedcontainer").
+			    jvmargs('-Xms16g -Xmx16g').
+			    search(Searching.new).
+			    docproc(DocumentProcessing.new).
+			    gateway(ContainerDocumentApi.new)).
+	  admin_metrics(Metrics.new).
+	  indexing("combinedcontainer").
+	  sd(selfdir + "doc.sd")
+  end
+
+  def test_reindexing_performance_and_impact
+    deploy_app(@app)
+    start
+
+    # First time is a dummy.
+    trigger_reindexing
+    wait_for_reindexing
+
+    qrserver = @vespa.container["combinedcontainer/0"]
+    document_count = 1 << 20
+
+    initial_file = dirs.tmpdir + "initial.json"
```

Comment:
I suggest moving all the data generation to a separate function and set the files as private variables with '@..'. This makes it easier to read to overall flow of the test.

Reply:
Can do. Yes, it's a bit verbose.